app/models/claim/matching_attribute_finder.rb

```diff
@@ -0,0 +1,41 @@
+class Claim
+  class MatchingAttributeFinder
+    ATTRIBUTES_TO_MATCH = [
+      "teacher_reference_number",
+      "email_address",
+      "national_insurance_number",
+      "bank_account_number",
+      "bank_sort_code",
+      "building_society_roll_number",
+    ].freeze
+
+    def initialize(source_claim, claims_to_compare = Claim.submitted)
+      @source_claim = source_claim
+      @claims_to_compare = claims_to_compare
+    end
+
+    def claim_ids_with_matching_attribute_keys
```

Comment:
`matching_attributes_keyed_by_claim_id` might make it clearer what are the keys and what are the values.